tests/behat/features/web/project-details/project_description.feature

```diff
@@ -41,6 +41,150 @@ Feature: Projects should have descriptions that can be changed by the project ow
+    And the element "#edit-description-submit-button" should be visible
+    Then I choose "French" from selector "#description-language-selector"
+    And I wait for AJAX to finish
+    Then I fill in "edit-description" with "This is a description translation"
+    And I click "#edit-description-submit-button"
+    And I wait for AJAX to finish
+    Then the element "#description" should be visible
+    And the element "#edit-description-ui" should not be visible
+    And I should see "my description"
+    When I click "#edit-description-button"
+    And I wait for AJAX to finish
+    Then the element "#edit-description" should be visible
+    And the element "#description-language-selector" should be visible
+    Then I choose "French" from selector "#description-language-selector"
+    And I wait for AJAX to finish
+    Then I should see "This is a description translation"
+
+  Scenario: Changing custom translation language and keeping changes
+    Given I log in as "Catrobat"
+    And I go to "/app/project/1"
+    And I wait for the page to be loaded
+    Then the element "#edit-description-button" should be visible
+    When I click "#edit-description-button"
+    And I wait for AJAX to finish
+    Then the element "#description" should not be visible
+    But the element "#edit-description" should be visible
+    And the element "#description-language-selector" should be visible
+    And the element "#edit-description-submit-button" should be visible
+    Then I choose "French" from selector "#description-language-selector"
+    And I wait for AJAX to finish
+    Then I fill in "edit-description" with "This is a description translation"
+    Then I choose "Russian" from selector "#description-language-selector"
+    And I should see "Would you like to keep your current changes?"
+    When I click ".swal2-confirm"
+    And I should see "This is a description translation"
+    And I should see "Russian"
+
+  Scenario: Changing custom translation language and discarding changes
+    Given I log in as "Catrobat"
+    And I go to "/app/project/1"
+    And I wait for the page to be loaded
+    Then the element "#edit-description-button" should be visible
+    When I click "#edit-description-button"
+    And I wait for AJAX to finish
+    Then the element "#description" should not be visible
+    But the element "#edit-description" should be visible
+    And the element "#description-language-selector" should be visible
+    And the element "#edit-description-submit-button" should be visible
+    Then I choose "French" from selector "#description-language-selector"
+    And I wait for AJAX to finish
+    Then I fill in "edit-description" with "This is a description translation"
+    Then I choose "Russian" from selector "#description-language-selector"
+    And I should see "Would you like to keep your current changes?"
+    When I click ".swal2-deny"
+    And I wait for AJAX to finish
+    And I should see "" in "edit-description"
```

Comment:
Looking that the test results, i don't think this empty string matched the code. Maybe you need a special implementation for empty string?

Reply:
Yeah, I knew there were a few problems with the tests when I first pushed them. I think they should be fixed now